File: ParkingLot/ParkingLot_Node/vehicles/vehicle.ts
export const VehicleType = {
  bike: "Bike",
  car: "Car",
  truck: "Truck",
};
export type VehicleType = (typeof VehicleType)[keyof typeof VehicleType];

export abstract class Vehicle {
  _license: string;
  _vehicleType: VehicleType;

  constructor(license: string, vehicleType: VehicleType) {
    this._license = license;
    this._vehicleType = vehicleType;
  }

  get license() {
    return this._license;
  }

  get vehicleType() {
    return this._vehicleType;
  }
}

File: ParkingLot/ParkingLot_Go/vehicles/vehicle.go
package vehicles

type VehicleType int

const (
	Bike VehicleType = iota
	Car
	Truck
)

type Vehicle interface {
	GetType() VehicleType
	GetLicense() string
}

type BaseVehicle struct {
	license     string
	vehicleType VehicleType
}

func (v *BaseVehicle) GetType() VehicleType {
	return v.vehicleType
}

func (v *BaseVehicle) GetLicense() string {
	return v.license
}
